extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -924,6 +975,61 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     return AD_STATE.INSERTED;
   }
 
+  /**
+   *
+   * @param {Element} adPageElement
+   */
+  maybeCreateAdChoices_(adPageElement) {
+    const {iframe} = this.lastCreatedAdImpl_;
+    const iframeDoc = getFrameDoc(iframe);
+    const href = getContentFromMetaTag(iframeDoc, Meta.ATTRIBUTION_URL);
+    const src = getContentFromMetaTag(iframeDoc, Meta.ATTRIBUTION_ICON);
+
+    // Ad Choices is optional, but need both to render.
+    if (!href && !src) {
+      return;
+    }
+
+    if (!href || !src) {
+      dev().warn('Both icon and URL must be supplied for Ad Choices.');
+      return;
+    }
+
+    const root = createElementWithAttributes(
+      this.doc_,
+      'div',
```

Comment:
If it acts as a button, please just use the `button` HTML tag. It'll make a11y easier once we finally prioritize it

Reply:
As long as you keep the `role="button"` it'll work too
Share icons should be `button` elements too.... :/

Reply:
Just tried this, apparently you cant attach a shadow root to a button :(
![image](https://user-images.githubusercontent.com/16087874/61494783-f6e04b00-a96b-11e9-862c-e3e203860f3d.png)